[ACM-7739]: Adding disconnected docs for bare metal and OpenShift Virtualization (#5548)
open-cluster-management/rhacm-docs: clusters/hosted_control_planes/dual_stack_deploy_registry.adoc

[#dual-stack-deploy-registry]
= Deploying a registry for a dual stack network

> For development environments, deploy a small, self-hosted registry by using a Podman container. For production environments, use a more reliable solution, such as Red Hat Quay, Nexus, or Artifactory.

Reviewer: I would remove "more reliable". 
```suggestion
For development environments, deploy a small, self-hosted registry by using a Podman container. For production environments, deploy an enterprise-hosted registry, such as Red Hat Quay, Nexus, or Artifactory.
```

Author: Fixed for this file and the ipv4 and ipv6 versions.

To deploy a small registry by using Podman, complete the following steps:

. As a privileged user, access the `${HOME}` directory and create the following script:

+

> ----

Reviewer: ```suggestion
[source,bash]
----
```

Author: fixed for dual stack, ipv4, and ipv6 versions of this file

if [[ ! -f $PULL_SECRET ]];then
  echo "Pull Secret not found, exiting..."
  exit 1
fi

dnf -y install podman httpd httpd-tools jq skopeo libseccomp-devel
export IP=$(ip -o addr show $PRIMARY_NIC | head -1 | awk '{print $4}' | cut -d'/' -f1)
REGISTRY_NAME=registry.$(hostname --long)
REGISTRY_USER=dummy
REGISTRY_PASSWORD=dummy
KEY=$(echo -n $REGISTRY_USER:$REGISTRY_PASSWORD | base64)
echo "{\"auths\": {\"$REGISTRY_NAME:5000\": {\"auth\": \"$KEY\", \"email\": \"sample-email@domain.ltd\"}}}" > /root/disconnected_pull.json
mv ${PULL_SECRET} /root/openshift_pull.json.old
jq ".auths += {\"$REGISTRY_NAME:5000\": {\"auth\": \"$KEY\",\"email\": \"sample-email@domain.ltd\"}}" < /root/openshift_pull.json.old > $PULL_SECRET
mkdir -p /opt/registry/{auth,certs,data,conf}
cat <<EOF > /opt/registry/conf/config.yml
version: 0.1
log:
  fields:
    service: registry
storage:
  cache:
    blobdescriptor: inmemory
  filesystem:
    rootdirectory: /var/lib/registry
  delete:
    enabled: true
http:
  addr: :5000
  headers:
    X-Content-Type-Options: [nosniff]
health:
  storagedriver:
    enabled: true
    interval: 10s
    threshold: 3
compatibility:
  schema1:
    enabled: true

> EOF

Reviewer: "EOF" should be removed. Is there a way to make this YAML more clean? We usually instruct users to create a YAML file and then apply it.

Author: I can ask engineering.

Author: @jparrill Typically in the official docs, we instruct users to create a YAML file and then apply it. You can see an example here: https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/clusters/cluster_mce_overview#hosted-control-planes-create-infraenv.

Can you help me clean up this code to better meet the doc requirements?

Author: @dockerymick As far as I understand it, because the "EOF" syntax is part of a script, I don't know how to rewrite it like I usually would in the docs. Usually, I see it in the context of creating a YAML file with specific contents and then applying the YAML file by running a command. Since the YAML is embedded in the script here, I think it might need to stay that way.

Reviewer 2: We don't normally document scripts and that is why we ask them to remove EOF from things like YAML file samples, so Mikela is right.

Author: @swopebe When we meet on Monday, I'd like to better understand why we don't normally document scripts. I'm not sure what to do in this case, as the script is an important part of this procedure. I understand why we want to present YAML in a cleaner, clearer way, but I don't know how to do it in this case. I'd appreciate some help!
